infra/modules/azure_app_service/locals.tf

```diff
@@ -15,9 +20,9 @@ locals {
   }
 
```

Comment:
what about something like this istead? the general concept is to use map wherever possible instead of ternaries which are harder to read and maintain:

```
# Tier mappings for better scalability and readability
 tier_mapping = {
   "test"     = "s"
   "standard" = "m"    
   "premium"  = "l"
   "xs"       = "xs"
   "s"        = "s"
   "m"        = "m"
   "l"        = "l"
   "xl"       = "xl"
 }

 # Default to the mapped tier or retain the original tier if not found
  
tier = lookup(local.tier_mapping, var.tier, var.tier)

 # Worker process count based on tier
 worker_process_count = {
   "s"  = 2
   "m"  = 4
   "l"  = 8
   "xl" = 8
   "xs" = 1
 }[local.tier]
```